packages/cli/src/ceramic-daemon.ts

```diff
@@ -180,34 +179,53 @@ class CeramicDaemon {
+    multiqueriesRouter.postAsync('/', this.multiQuery.bind(this))
+    streamsRouter.getAsync('/:streamid', this.state.bind(this))
+    pinsRouter.getAsync('/:streamid', this.listPinned.bind(this))
+    pinsRouter.getAsync('/', this.listPinned.bind(this))
+    nodeRouter.getAsync('/chains', this.getSupportedChains.bind(this))
+    nodeRouter.getAsync('/healthcheck', this.healthcheck.bind(this))
+    documentsRouter.getAsync('/:docid', this.stateOld.bind(this)) // Deprecated
+    recordsRouter.getAsync('/:streamid', this.commits.bind(this)) // Deprecated
 
     if (!gateway) {
-      app.postAsync(toApiPath('/streams'), this.createStreamFromGenesis.bind(this))
-      app.postAsync(toApiPath('/commits'), this.applyCommit.bind(this))
-      app.postAsync(toApiPath('/pins/:streamid'), this.pinStream.bind(this))
-      app.deleteAsync(toApiPath('/pins/:streamid'), this.unpinStream.bind(this))
+      streamsRouter.postAsync('/', this.createStreamFromGenesis.bind(this))
+      commitsRouter.postAsync('/', this.applyCommit.bind(this))
+      pinsRouter.postAsync('/:streamid', this.pinStream.bind(this))
+      pinsRouter.deleteAsync('/:streamid', this.unpinStream.bind(this))
 
-      app.postAsync(toApiPath('/documents'), this.createDocFromGenesis.bind(this)) // Deprecated
-      app.postAsync(toApiPath('/records'), this.applyCommit.bind(this)) // Deprecated
+      documentsRouter.postAsync('/', this.createDocFromGenesis.bind(this)) // Deprecated
+      recordsRouter.postAsync('/', this.applyCommit.bind(this)) // Deprecated
     } else {
-      app.postAsync(toApiPath('/streams'), this.createReadOnlyStreamFromGenesis.bind(this))
-      app.postAsync(toApiPath('/commits'),  this._notSupported.bind(this))
-      app.postAsync(toApiPath('/pins/:streamid'),  this._notSupported.bind(this))
-      app.deleteAsync(toApiPath('/pins/:streamid'),  this._notSupported.bind(this))
+      streamsRouter.postAsync('/', this.createReadOnlyStreamFromGenesis.bind(this))
+      commitsRouter.postAsync('/',  this._notSupported.bind(this))
+      pinsRouter.postAsync('/:streamid',  this._notSupported.bind(this))
+      pinsRouter.deleteAsync('/:streamid',  this._notSupported.bind(this))
 
-      app.postAsync(toApiPath('/documents'), this.createReadOnlyDocFromGenesis.bind(this)) // Deprecated
-      app.postAsync(toApiPath('/records'),  this._notSupported.bind(this)) // Deprecated
+      documentsRouter.postAsync('/', this.createReadOnlyDocFromGenesis.bind(this)) // Deprecated
+      recordsRouter.postAsync('/',  this._notSupported.bind(this)) // Deprecated
     }
+
+    baseRouter.use('/commits', commitsRouter)
+    baseRouter.use('/documents', documentsRouter)
+    baseRouter.use('/multiqueries', multiqueriesRouter)
+    baseRouter.use('/node', nodeRouter)
+    baseRouter.use('/pins', pinsRouter)
+    baseRouter.use('/records', recordsRouter)
+    baseRouter.use('/streams', streamsRouter)
+    baseRouter.use(errorHandler(this.diagnosticsLogger))
```

Comment:
@valmack This line should not be necessary.

Reply:
I guess you are saying line 226 with the errorHandler?

But it actually needs to be added to each individual router to properly parse params as well. If we don't use it in these routers our "requests per endpoint" will exclude failed requests. So I'm actually going to add it in more places.

With
```
original_url=/api/v0/documents/k2t6wyfsu4pg2r88kogtioocim6p6ewxnhakzau0ekd4xmparhncekviu7nvua/
base_url=/api/v0/documents
path=/k2t6wyfsu4pg2r88kogtioocim6p6ewxnhakzau0ekd4xmparhncekviu7nvua/
params=-
```

Without
```
original_url=/api/v0/documents/k2t6wyfsu4pg2r88kogtioocim6p6ewxnhakzau0ekd4xmparhncekviu7nvua/
base_url=/api/v0
path=/documents/k2t6wyfsu4pg2r88kogtioocim6p6ewxnhakzau0ekd4xmparhncekviu7nvua/
params=-
```

Reply:
Wow. That's sad and concerning.